core_url.go

```diff
@@ -54,7 +54,7 @@ func (ub *urlBuilder) Clone() *urlBuilder {
 func (ub *urlBuilder) URL() (u *url.URL, err error) {
 	u, err = url.Parse(ub.baseurl)
 	if err != nil {
-		return new(url.URL), err
+		return nil, err
```

Comment:
This introduces a bug. The docs specify that u will never be nil.

Reply:
you can also return `u`

Reply:
`url.Parse` does return a nil `url.URL` for errors.